example/router/router.cpp

```diff
@@ -0,0 +1,328 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+//[example_router
+
+/*
+    This example defines a router for URL paths.
+    Each path is associated with a callback
+    function.
+*/
+
+#include <router.hpp>
+#include <src.hpp>
+
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/config.hpp>
+
+#include <iostream>
+#include <functional>
+
+/*
+ * Aliases
+ */
+
+namespace urls = boost::urls;
+namespace asio = boost::asio;
+namespace beast = boost::beast;
+namespace http = beast::http;
+using string_view = urls::string_view;
+using match_results = urls::router_base::match_results_base;
```

Comment:
this type is uuuuuuugly

Reply:
Yes. We had a kind of problem here. 

The ideal type would be `router_type::match_results` but I can't do that because it doesn't exist yet. :laughing:

Reply:
hmmm I will have to think about that, I believe we can solve it